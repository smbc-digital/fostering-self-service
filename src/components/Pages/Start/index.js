import React, { Fragment, useContext } from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'
import { TaskItem, TaskStatus } from 'smbc-react-components'
import { Context } from '../../../context'

const TaskLink = ({ route, name, status }) => {

	const renderStatus = () => {
		switch (status) {
			case TaskStatus.Completed:
				return <span className='status-completed'>Completed</span>
			case TaskStatus.NotCompleted:
				return <span className='status-not-completed'>Not completed</span>
			default:
				return
		}
	}

	return <div className='anchor-with-status'>
		<Link to={route}>{name}</Link>
		{renderStatus()}
	</div>
}

const FormLinks = () => {
	const { statuses: {
		//childrenLivingAwayFromYourHomeStatus,
		languageSpokenInYourHomeStatus,
		//tellUsAboutYourInterestInFosteringStatus,
		tellUsAboutYourselfStatus,
		yourEmploymentDetailsStatus,
		//yourFosteringHistoryStatus,
		//yourHealthStatus,
		//yourHouseholdStatus,
		yourPartnershipStatus
	} } = useContext(Context)
	return <Fragment>
		<p>To help prepare for your home visit, you can answer the questions in this section. This step is optional, however your fostering journey can be made simpler by telling us more about yourself. The questions will give you an idea of the conversation that you’ll have during your home visit and the social worker will be able to see your answers beforehand.</p>
		<TaskLink route='/fostering/known-by-another-name' status={tellUsAboutYourselfStatus} name='Tell us more about you' />
		<TaskLink route='/fostering/are-you-employed' status={yourEmploymentDetailsStatus} name='Your employment details' />
		<TaskLink route='#' status={languageSpokenInYourHomeStatus} name='Tell us more about languages that are spoken in your home' />
		<TaskLink route='#' status={yourPartnershipStatus} name='Your partnership status' />
	</Fragment>
}

const Start = () => {
	const tasks = [
		{
			title: 'Contact the fostering team',
			body: () => <p>Contact the fostering team to tell us that you’re interested in becoming a foster carer.</p>,
			status: 0
		},
		{
			title: 'Find out more about fostering',
			body: () => <p>Speak to the fostering team about your interest in fostering and arrange a home visit.</p>,
			status: 1
		},
		{
			title: 'Answer questions before your home visit',
			body: () => <FormLinks />,
			status: 1
		},
		{
			title: 'Home visit',
			body: () => <p>The social worker will come to your home to find out more about you and your reasons for wanting to become a foster carer. If you’ve answered the questions in the previous section, you’ll talk more about them. The home visit stage can be completed in one session but it may require more sessions to make sure that the social worker has all of the information that they need.</p>,
			disabled: true
		},
		{
			title: 'Additional information',
			body: () => <p>If you’ve been successful at your home visit, you’ll need to give us some more information so that you can move onto the next stage of your fostering journey.</p>,
			status: 2,
			disabled: true
		},
		{
			title: 'Assessment and training',
			body: () => <p>You’ll complete the assessment stage of your fostering journey and attend training sessions.</p>,
			status: 2,
			disabled: true
		},
		{
			title: 'Approval',
			body: () => <p>You’ll go to panel and a decision will be made about whether you can become a foster carer.</p>,
			disabled: true
		}
	]

	return (
		<Fragment>
			<h1>Start your fostering journey</h1>
			<ol className='task-item-list'>
				{tasks.map((task, index) => <li
					key={index}
					className={task.disabled ? 'disabled' : undefined}>
					<TaskItem
						key={index}
						title={task.title}
						body={task.body}
						status={task.status}
					/>
				</li>)}
			</ol>
		</Fragment>
	)
}

Start.propTypes = {
	history: PropTypes.object
}

TaskLink.propTypes = {
	route: PropTypes.string,
	name: PropTypes.string,
	status: PropTypes.object
}

export default Start
