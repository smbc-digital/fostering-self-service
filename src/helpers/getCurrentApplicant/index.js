import { Applicant, SECOND_APPLICANT_ROUTE } from 'constants'

const getCurrentApplicant = match => {
    if (match.params && match.params[0] === SECOND_APPLICANT_ROUTE) {
        return Applicant.SecondApplicant
    }

    return Applicant.FirstApplicant
}

export default getCurrentApplicant