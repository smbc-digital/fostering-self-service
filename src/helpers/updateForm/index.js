import { fetchWithTimeout } from '../index'

const TaskStatus = {
    None: 0,
    Completed: 1,
    NotCompleted: 2,
    CantStart: 3
}

const StageName = {
    HomeVisit: 0,
    Application: 1
}

const HomeVisitFormName = {
    TellUsAboutYourself: 0,
    YourEmploymentDetails: 1,
    LanguagesSpokenInYourHome: 2,
    YourPartnership: 3,
    YourFosteringHistory: 4,
    YourHealth: 5,
    TellUsAboutYourInterestInFostering: 6,
    YourHousehold: 7,
    ChildrenLivingAwayFromYourHome: 8
}

const ApplicationFormName = {
    References: 0,
    GpDetails: 1,
    AddressHistory: 2
}

const getHomeVisitUpdateEndpoint = form => {
    switch (form) {
        case HomeVisitFormName.TellUsAboutYourself:
            return '/fostering/home-visit/about-yourself'
        case HomeVisitFormName.YourEmploymentDetails:
            return '/fostering/home-visit/your-employment-details'
        case HomeVisitFormName.LanguagesSpokenInYourHome:
            return '/fostering/home-visit/languages-spoken-in-your-home'
        case HomeVisitFormName.YourFosteringHistory:
            return '/fostering/home-visit/your-fostering-history'
        case HomeVisitFormName.YourPartnership:
            return '/fostering/home-visit/partnership-status'
        case HomeVisitFormName.TellUsAboutYourInterestInFostering:
            return '/fostering/home-visit/interest-in-fostering'
        case HomeVisitFormName.YourHealth:
            return '/fostering/home-visit/about-your-health'
        case HomeVisitFormName.YourHousehold:
            return '/fostering/home-visit/household'
        case HomeVisitFormName.ChildrenLivingAwayFromYourHome:
            return '/fostering/home-visit/children-living-away-from-home'
        default:
            throw new Error('No matching endpoint for given form.')
    }
}

const getApplicationUpdateEndpoint = form => {
    switch (form) {
        case ApplicationFormName.References:
            return '/fostering/application/references'
        case ApplicationFormName.GpDetails:
            return '/fostering/application/gp-details'
        default:
            throw new Error('No matching endpoint for given form.')
    }
}

const getFormStageEndpoint = stage => {
    switch (stage) {
        case StageName.HomeVisit: 
            return '/fostering/home-visit'
        case StageName.Application:
            return '/fostering/application'
        default:
            throw new Error('No matching endpoint for given stage.')
    }
}

const reduceProperties = object => Object.keys(object).reduce((acc, property) => {
    return {
        ...acc,
        [property]: object[property].value
    }
}, {})

const updateForm = async (endpoint, formData) => {
    const parsedFormData = parseFormData(formData)

    const response = await fetchWithTimeout(endpoint, {
        method: 'PATCH',
        credentials: 'include',
        body: JSON.stringify(parsedFormData),
        headers: {
            'Accept': 'application/json',
            'Content-Type': 'application/json'
        }
    }, 30000)

    if (!response.ok) {
        throw Error(response.statusText)
    }

    return await response.json()
}

const updateFormStatus = ({ form, stage, currentStatus, setStatus }) => {
    if (currentStatus === TaskStatus.None) {
        fetchWithTimeout(`${getFormStageEndpoint(stage)}/status`,
            {
                method: 'PATCH',
                credentials: 'include',
                body: JSON.stringify({
                    status: TaskStatus.NotCompleted,
                    form
                }),
                headers: {
                    'Accept': 'application/json',
                    'Content-Type': 'application/json'
                }
            }, 30000)

        setStatus(TaskStatus.NotCompleted)
    }
}

const parseFormData = ({ firstApplicant, secondApplicant, familyReference, firstPersonalReference, secondPersonalReference, ...formData }) => {
    let parsedObject = reduceProperties(formData)

    if (firstApplicant) {
        parsedObject.firstApplicant = reduceProperties(firstApplicant)
    }

    if (secondApplicant) {
        parsedObject.secondApplicant = reduceProperties(secondApplicant)
    }

    if (familyReference) {
        parsedObject.familyReference = reduceProperties(familyReference)
        parsedObject.familyReference.address = familyReference.address.value
    }

    if (firstPersonalReference) {
        parsedObject.firstPersonalReference = reduceProperties(firstPersonalReference)
        parsedObject.firstPersonalReference.address = firstPersonalReference.address.value
    }

    if (secondPersonalReference) {
        parsedObject.secondPersonalReference = reduceProperties(secondPersonalReference)
        parsedObject.secondPersonalReference.address = secondPersonalReference.address.value
    }

    return parsedObject
}

const updateApplicationForm = async (form, formData) => {
    const endpoint = getApplicationUpdateEndpoint(form)

    return await updateForm(endpoint, formData)
}

const updateHomeVisitForm = async (form, formData) => {
    const endpoint = getHomeVisitUpdateEndpoint(form)

    return await updateForm(endpoint, formData)
}

export {
    TaskStatus,
    StageName,
    HomeVisitFormName,
    ApplicationFormName,
    updateFormStatus,
    parseFormData,
    updateApplicationForm,
    updateHomeVisitForm
}