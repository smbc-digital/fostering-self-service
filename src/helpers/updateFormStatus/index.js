export const TaskStatus = {
    None: 0,
    Completed: 1,
    NotCompleted: 2,
    CantStart: 3
}

export const FormName =
{
    TellUsAboutYourself: 0,
    YourEmploymentDetails: 1,
    LanguageSpokenInYourHome: 2,
    YourPartnership: 3,
    YourFosteringHistory: 4,
    YourHealth: 5,
    TellUsAboutYourInterestInFostering: 6,
    YourHousehold: 7,
    ChildrenLivingAwayFromYourHome: 7
}

export default function (form, currentStatus, setStatus) {
    if (currentStatus === TaskStatus.None) {
        fetch('/front-end-endpoint-update-status',
            {
                method: 'PATCH',
                credentials: 'include',
                body: JSON.stringify({
                    status: TaskStatus.NotCompleted,
                    form
                })
            })

        setStatus(TaskStatus.NotCompleted)
    }
}