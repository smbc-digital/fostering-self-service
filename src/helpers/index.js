import fetchWithTimeout from './fetchWithTimeout'
import { getPageRoute } from './pagehelper'
import getCurrentApplicant from './getCurrentApplicant'
import { updateFormStatus, updateForm, FormName, TaskStatus } from './updateForm'

export {
    fetchWithTimeout,
    getPageRoute,
    getCurrentApplicant,
    updateFormStatus,
    updateForm,
    FormName,
    TaskStatus
}