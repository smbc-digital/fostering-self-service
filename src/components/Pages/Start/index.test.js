import moment from 'moment-timezone'
import { React, mount, useContextMock, renderer } from 'helpers/SetupTest'
import { BrowserRouter as Router } from 'react-router-dom'
import Start from './index'

describe('Start', () => {

    beforeEach(() => {
        window.HTMLElement.prototype.scrollIntoView = jest.fn()

        useContextMock.mockReturnValue({
            homeVisitDateTime: {
                value: moment().add(10, 'minutes').format('DD/MM/YYYY HH:mm')
            },
            enableAdditionalInformationSection: {
                value: true
            },
            statuses: {
                tellUsAboutYourselfStatus: 1,
                yourEmploymentDetailsStatus: 2,
                languageSpokenInYourHomeStatus: 1,
                yourPartnershipStatus: 1,
                yourFosteringHistoryStatus: 2,
            },
            secondApplicant: {
                
            }
        })
    })
    
    it('should render correct Task status', () => {
        var wrapper = mount(<Router><Start/></Router>)
       
        expect(wrapper.find('span').get(0).props.className).toEqual('status-completed')
        expect(wrapper.find('span').get(1).props.className).toEqual('status-completed')
        expect(wrapper.find('span').get(2).props.className).toEqual('status-completed')
    })

    it('should render disabled task list items', () => {
        // Arrange
        useContextMock.mockReturnValue({
            statuses: {},
            secondApplicant: {},
            homeVisitDateTime: { 
                value: moment().add(30, 'minutes').format('DD/MM/YYYY HH:mm')
            },
            enableAdditionalInformationSection: {
                value: true
            },
        })

        // Act
        let wrapper = mount(<Router><Start /></Router>)

        // Assert
        wrapper.find('FormLinks').children('TaskLink').forEach(node => {
            expect(node.props().disabled).toBe(true)
        })
    })

    it('should render task list items', () => {
        // Arrange
        useContextMock.mockReturnValue({
            homeVisitDateTime: {
                value: moment().add(10, 'd').format('DD/MM/YYYY HH:mm')
            },
            enableAdditionalInformationSection: {
                value: false
            },
            statuses: {}
        })

        // Act
        let wrapper = mount(<Router><Start /></Router>)

        // Assert
        wrapper.find('FormLinks').children('TaskLink').forEach(node => {
            expect(node.props().disabled).toBe(false)
        })
    })
    
    it('should render correct Task status after home visit is completed', () => {
        useContextMock.mockReturnValue({
            homeVisitDateTime: {
                value: moment().subtract(10, 'minutes').format('DD/MM/YYYY HH:mm')
            },
            enableAdditionalInformationSection: {
                value: false
            },
            statuses: {
                tellUsAboutYourselfStatus: 1,
                yourEmploymentDetailsStatus: 1,
                languageSpokenInYourHomeStatus: 1,
                yourPartnershipStatus: 1,
                yourFosteringHistoryStatus: 1,              
            },
        })

        var wrapper = mount(<Router><Start/></Router>)
        expect(wrapper.find('TaskItem').get(2).props.status).toEqual(1)
    })

    it('should render correct status for Home Visit section', () => {
        // Arrange
        useContextMock.mockReturnValue({
            homeVisitDateTime: {
                value: moment().subtract(10, 'minutes').format('DD/MM/YYYY HH:mm')
            },
            enableAdditionalInformationSection: {
                value: true
            },
            statuses: {
                tellUsAboutYourselfStatus: 1,
                yourEmploymentDetailsStatus: 1,
                languageSpokenInYourHomeStatus: 1,
                yourPartnershipStatus: 1,
                yourFosteringHistoryStatus: 1,
            },
        })
        var wrapper = mount(<Router><Start/></Router>)
        
        // Assert
        expect(wrapper.find('TaskItem').get(3).props.status).toEqual(1)
    })


    it('should not render TaskLink statuses', () => {
        // Arrange
        useContextMock.mockReturnValue({
            homeVisitDateTime: {
                value: moment().subtract(10, 'minutes').format('DD/MM/YYYY HH:mm')
            },
            enableAdditionalInformationSection: {
                value: true
            },
            statuses: {
                tellUsAboutYourselfStatus: 1,
                yourEmploymentDetailsStatus: 1,
                languageSpokenInYourHomeStatus: 1,
                yourPartnershipStatus: 1,
                yourFosteringHistoryStatus: 1,
            },
        })
        var wrapper = mount(<Router><Start/></Router>)


        // Assert
        wrapper.find('TaskLink').forEach(item => {
            expect(item.props().status).toBe(undefined)
        })
    })

    it('should render home visit date and time', () => {
        //Arrange
        const mockedDate = moment()
        useContextMock.mockReturnValue({
            homeVisitDateTime: {
                value: mockedDate.format('DD/MM/YYYY HH:mm')
            },
            enableAdditionalInformationSection: {
                value: true
            },
            statuses: {
                tellUsAboutYourselfStatus: 1,
                yourEmploymentDetailsStatus: 1,
                languageSpokenInYourHomeStatus: 1,
                yourPartnershipStatus: 1,
                yourFosteringHistoryStatus: 1,
            }
        })

        //Act
        const wrapper = mount(<Router><Start/></Router>)

        //Assert
        expect(wrapper.find('TaskItem').at(3).text()).toContain(mockedDate.format('DD/MM/YYYY'))
        expect(wrapper.find('TaskItem').at(3).text()).toContain(mockedDate.format('HH:mm'))
    })

    describe('snapshot', () => {
        it('renders correctly', () => {
            useContextMock.mockReturnValue({
                homeVisitDateTime: {
                    value: null
                },
                enableAdditionalInformationSection: {
                    value: true
                },
                statuses: {
                    tellUsAboutYourselfStatus: 1,
                    yourEmploymentDetailsStatus: 2,
                    languageSpokenInYourHomeStatus: 1,
                    yourPartnershipStatus: 1,
                    yourFosteringHistoryStatus: 2,
                },
                secondApplicant: {
                
                }
            })

            const tree = renderer
                .create(<Router><Start/></Router>)
                .toJSON()

            expect(tree).toMatchSnapshot()
        })

        it('renders correctly with home visit date', () => {
            useContextMock.mockReturnValue({
                homeVisitDateTime: {
                    value: moment('25/07/2190 13:10', 'DD/MM/YYYY HH:mm')
                },
                enableAdditionalInformationSection: {
                    value: true
                },
                statuses: {
                    tellUsAboutYourselfStatus: 1,
                    yourEmploymentDetailsStatus: 2,
                    languageSpokenInYourHomeStatus: 1,
                    yourPartnershipStatus: 1,
                    yourFosteringHistoryStatus: 2,
                },
                secondApplicant: {
                
                }
            })

            const tree = renderer
                .create(<Router><Start/></Router>)
                .toJSON()

            expect(tree).toMatchSnapshot()
        })
    })
})