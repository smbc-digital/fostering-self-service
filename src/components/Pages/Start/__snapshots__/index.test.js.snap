// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start snapshot renders correctly 1`] = `
Array [
  <h1>
    Start your fostering journey
  </h1>,
  <ol
    className="task-item-list"
  >
    <li
      className={undefined}
    >
      <div
        className="task-item"
      >
        <div
          className="task-item-heading"
        >
          <span
            className="status-completed"
          >
            Completed
          </span>
          <h2>
            Contact the fostering team
          </h2>
        </div>
        <p>
          Contact the fostering team to tell us that you’re interested in becoming a foster carer.
        </p>
        <hr />
      </div>
    </li>
    <li
      className={undefined}
    >
      <div
        className="task-item"
      >
        <div
          className="task-item-heading"
        >
          <span
            className="status-completed"
          >
            Completed
          </span>
          <h2>
            Find out more about fostering
          </h2>
        </div>
        <p>
          Speak to the fostering team about your interest in fostering and arrange a home visit.
        </p>
        <hr />
      </div>
    </li>
    <li
      className={undefined}
    >
      <div
        className="task-item"
      >
        <div
          className="task-item-heading"
        >
          <h2>
            Answer questions before your home visit
          </h2>
        </div>
        <p>
          You can answer the questions in this section to help you to prepare for your home visit. This step is optional however, your fostering journey can be made simpler by telling us more about yourself. You can make changes to your answers up to 30 minutes before your home visit and your social worker will be able to see your answers before your appointment.
        </p>
        <div
          className="anchor-with-status"
        >
          <a
            href="/fostering/known-by-another-name"
            onClick={[Function]}
          >
            Tell us more about you
          </a>
          <span
            className="status-completed"
          >
            Completed
          </span>
        </div>
        <div
          className="anchor-with-status"
        >
          <a
            href="/fostering/are-you-employed"
            onClick={[Function]}
          >
            Your employment details
          </a>
          <span
            className="status-not-completed"
          >
            Not completed
          </span>
        </div>
        <div
          className="anchor-with-status"
        >
          <a
            href="/fostering/languages-spoken-in-your-home"
            onClick={[Function]}
          >
            Tell us about the languages that are spoken in your home
          </a>
          <span
            className="status-completed"
          >
            Completed
          </span>
        </div>
        <div
          className="anchor-with-status"
        >
          <a
            href="/fostering/your-partnership-status"
            onClick={[Function]}
          >
            Your partnership status
          </a>
          <span
            className="status-completed"
          >
            Completed
          </span>
        </div>
        <div
          className="anchor-with-status"
        >
          <a
            href="/fostering/your-fostering-history"
            onClick={[Function]}
          >
            Your fostering history
          </a>
          <span
            className="status-not-completed"
          >
            Not completed
          </span>
        </div>
        <div
          className="anchor-with-status"
        >
          <a
            href="/fostering/about-your-health"
            onClick={[Function]}
          >
            Your health
          </a>
        </div>
        <div
          className="anchor-with-status"
        >
          <a
            href="/fostering/your-interest-in-fostering"
            onClick={[Function]}
          >
            Tell us about your interest in fostering
          </a>
        </div>
        <div
          className="anchor-with-status"
        >
          <a
            href="/fostering/your-household"
            onClick={[Function]}
          >
            Your household
          </a>
        </div>
        <div
          className="anchor-with-status"
        >
          <a
            href="/fostering/children-under-sixteen-living-away"
            onClick={[Function]}
          >
            Children living away from your home
          </a>
        </div>
      </div>
    </li>
    <li
      className="disabled"
    >
      <div
        className="task-item"
      >
        <div
          className="task-item-heading"
        >
          <h2>
            Home visit
          </h2>
        </div>
        <p>
          Your social worker will come to your home to find out more about you and your reasons for wanting to become a foster carer. If you’ve answered the questions in section 3, you’ll talk more about them. 
        </p>
        <hr />
      </div>
    </li>
    <li
      className="disabled"
    >
      <div
        className="task-item"
      >
        <div
          className="task-item-heading"
        >
          <span
            className="status-cant-start"
          >
            Can't start yet
          </span>
          <h2>
            Additional information
          </h2>
        </div>
        <p>
          After your home visit, you must give us more information so that we can carry out personal checks and contact your referees.
        </p>
        <div
          className="anchor-with-status"
        >
          <p>
            Address history
          </p>
        </div>
        <div
          className="anchor-with-status"
        >
          <p>
            Your GP details
          </p>
        </div>
        <div
          className="anchor-with-status"
        >
          <p>
            Personal references
          </p>
        </div>
        <div
          className="anchor-with-status"
        >
          <p>
            Personal relationships with local councillors or council employees
          </p>
        </div>
      </div>
    </li>
    <li
      className="disabled"
    >
      <div
        className="task-item"
      >
        <div
          className="task-item-heading"
        >
          <h2>
            Next steps
          </h2>
        </div>
        <p>
          The fostering team will use the information that you’ve provided to carry out statutory checks and request personal and health references from your referees and GP.
        </p>
        <p>
          You’ll complete the assessment stage of your fostering journey where you’ll:
        </p>
        <ul>
          <li>
            attend training sessions, including the Skills to Foster course to help prepare you for becoming a foster carer
          </li>
          <li>
            have more meetings with your social worker so that they can find out more about you and your reasons for wanting to become a foster carer
          </li>
          <li>
            get to know current foster carers and other people who are starting their fostering journey
          </li>
        </ul>
        <p>
          You’ll then go to panel where a decision will be made about whether you can become a foster carer.
        </p>
      </div>
    </li>
  </ol>,
]
`;

exports[`Start snapshot renders correctly with home visit date 1`] = `
Array [
  <h1>
    Start your fostering journey
  </h1>,
  <ol
    className="task-item-list"
  >
    <li
      className={undefined}
    >
      <div
        className="task-item"
      >
        <div
          className="task-item-heading"
        >
          <span
            className="status-completed"
          >
            Completed
          </span>
          <h2>
            Contact the fostering team
          </h2>
        </div>
        <p>
          Contact the fostering team to tell us that you’re interested in becoming a foster carer.
        </p>
        <hr />
      </div>
    </li>
    <li
      className={undefined}
    >
      <div
        className="task-item"
      >
        <div
          className="task-item-heading"
        >
          <span
            className="status-completed"
          >
            Completed
          </span>
          <h2>
            Find out more about fostering
          </h2>
        </div>
        <p>
          Speak to the fostering team about your interest in fostering and arrange a home visit.
        </p>
        <hr />
      </div>
    </li>
    <li
      className={undefined}
    >
      <div
        className="task-item"
      >
        <div
          className="task-item-heading"
        >
          <h2>
            Answer questions before your home visit
          </h2>
        </div>
        <p>
          You can answer the questions in this section to help you to prepare for your home visit. This step is optional however, your fostering journey can be made simpler by telling us more about yourself. You can make changes to your answers up to 30 minutes before your home visit and your social worker will be able to see your answers before your appointment.
        </p>
        <div
          className="anchor-with-status"
        >
          <a
            href="/fostering/known-by-another-name"
            onClick={[Function]}
          >
            Tell us more about you
          </a>
          <span
            className="status-completed"
          >
            Completed
          </span>
        </div>
        <div
          className="anchor-with-status"
        >
          <a
            href="/fostering/are-you-employed"
            onClick={[Function]}
          >
            Your employment details
          </a>
          <span
            className="status-not-completed"
          >
            Not completed
          </span>
        </div>
        <div
          className="anchor-with-status"
        >
          <a
            href="/fostering/languages-spoken-in-your-home"
            onClick={[Function]}
          >
            Tell us about the languages that are spoken in your home
          </a>
          <span
            className="status-completed"
          >
            Completed
          </span>
        </div>
        <div
          className="anchor-with-status"
        >
          <a
            href="/fostering/your-partnership-status"
            onClick={[Function]}
          >
            Your partnership status
          </a>
          <span
            className="status-completed"
          >
            Completed
          </span>
        </div>
        <div
          className="anchor-with-status"
        >
          <a
            href="/fostering/your-fostering-history"
            onClick={[Function]}
          >
            Your fostering history
          </a>
          <span
            className="status-not-completed"
          >
            Not completed
          </span>
        </div>
        <div
          className="anchor-with-status"
        >
          <a
            href="/fostering/about-your-health"
            onClick={[Function]}
          >
            Your health
          </a>
        </div>
        <div
          className="anchor-with-status"
        >
          <a
            href="/fostering/your-interest-in-fostering"
            onClick={[Function]}
          >
            Tell us about your interest in fostering
          </a>
        </div>
        <div
          className="anchor-with-status"
        >
          <a
            href="/fostering/your-household"
            onClick={[Function]}
          >
            Your household
          </a>
        </div>
        <div
          className="anchor-with-status"
        >
          <a
            href="/fostering/children-under-sixteen-living-away"
            onClick={[Function]}
          >
            Children living away from your home
          </a>
        </div>
      </div>
    </li>
    <li
      className="disabled"
    >
      <div
        className="task-item"
      >
        <div
          className="task-item-heading"
        >
          <h2>
            Home visit
          </h2>
        </div>
        <p>
          Your home visit appointment will be on
          <strong>
             
            25/07/2190
          </strong>
           at 
          <strong>
            13:10
          </strong>
        </p>
        <p>
          Your social worker will come to your home to find out more about you and your reasons for wanting to become a foster carer. If you’ve answered the questions in section 3, you’ll talk more about them. 
        </p>
        <hr />
      </div>
    </li>
    <li
      className="disabled"
    >
      <div
        className="task-item"
      >
        <div
          className="task-item-heading"
        >
          <span
            className="status-cant-start"
          >
            Can't start yet
          </span>
          <h2>
            Additional information
          </h2>
        </div>
        <p>
          After your home visit, you must give us more information so that we can carry out personal checks and contact your referees.
        </p>
        <div
          className="anchor-with-status"
        >
          <p>
            Address history
          </p>
        </div>
        <div
          className="anchor-with-status"
        >
          <p>
            Your GP details
          </p>
        </div>
        <div
          className="anchor-with-status"
        >
          <p>
            Personal references
          </p>
        </div>
        <div
          className="anchor-with-status"
        >
          <p>
            Personal relationships with local councillors or council employees
          </p>
        </div>
      </div>
    </li>
    <li
      className="disabled"
    >
      <div
        className="task-item"
      >
        <div
          className="task-item-heading"
        >
          <h2>
            Next steps
          </h2>
        </div>
        <p>
          The fostering team will use the information that you’ve provided to carry out statutory checks and request personal and health references from your referees and GP.
        </p>
        <p>
          You’ll complete the assessment stage of your fostering journey where you’ll:
        </p>
        <ul>
          <li>
            attend training sessions, including the Skills to Foster course to help prepare you for becoming a foster carer
          </li>
          <li>
            have more meetings with your social worker so that they can find out more about you and your reasons for wanting to become a foster carer
          </li>
          <li>
            get to know current foster carers and other people who are starting their fostering journey
          </li>
        </ul>
        <p>
          You’ll then go to panel where a decision will be made about whether you can become a foster carer.
        </p>
      </div>
    </li>
  </ol>,
]
`;
